### PR TITLE
tests: set/unset scenarios for logMetric 

### PR DESCRIPTION
**Pull request by acpana**
Adding step based testing for logMetric. Checking both the DCL results and direct results.

### CODE REVIEW

File: tests/e2e/testdata/scenarios/direct/llm_set_unset/basic/_object02.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    cnrm.cloud.google.com/state-into-spec: merge
+  finalizers:
+  - cnrm.cloud.google.com/finalizer
+  - cnrm.cloud.google.com/deletion-defender
+  generation: 5
+  name: logginglogmetric-${uniqueId}
+  namespace: ${projectId}
+spec:
+  description: an e2e test
+  disabled: true
```
Comment (acpana) [inline tests/e2e/testdata/scenarios/direct/llm_set_unset/basic/_object02.yaml:17]: I would've expected these to be unset 🤔

Comment (acpana) [reply tests/e2e/testdata/scenarios/direct/llm_set_unset/basic/_object02.yaml]: They are after setting the `state-into-spec: absent` annotation.

File: tests/e2e/script_test.go [hunk 2]
```diff
@@ -228,18 +231,21 @@ func TestE2EScript(t *testing.T) {
 				}
 
 				if os.Getenv("GOLDEN_REQUEST_CHECKS") != "" || os.Getenv("WRITE_GOLDEN_OUTPUT") != "" {
-					x := NewNormalizer(uniqueID, project)
+					if os.Getenv("KCC_USE_DIRECT_RECONCILERS") != "" && os.Getenv("GOLDEN_REQUEST_CHECKS") != "" {
+						h.Logf("ignoring http log checking for now for direct resources")
```
Comment (acpana) [inline tests/e2e/script_test.go:235]: I do have a plan to check the golden logs but the change may be a bit more involved.

AFAICT, when we check two golden logs:

- `non_direct_http.log` [1]
- `direct_http.log` [2]

The following needs to be checked at least:
- we want to check that a POST/ PUT  in [2] exists in [1] AND ignore any GETs in [1] before it.
- we want to check that a GET in [2] has at least one corresponding GET in [1].

This excludes some further normalization like ignoring user agents.

IMO I think we get a lot of validation from checking the KRM objects themselves already.

File: tests/e2e/testdata/scenarios/direct/llm_set_unset/basic/_object02.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    cnrm.cloud.google.com/project-id: ${projectId}
+    cnrm.cloud.google.com/state-into-spec: absent
+  finalizers:
+  - cnrm.cloud.google.com/finalizer
+  - cnrm.cloud.google.com/deletion-defender
+  generation: 4
+  name: logginglogmetric-${uniqueId}
+  namespace: ${projectId}
+spec:
+  filter: resource.type=gae_app AND severity<=INFO
```
Comment (acpana) [inline tests/e2e/testdata/scenarios/direct/llm_set_unset/basic/_object02.yaml:30]: notice `disabled` and `description` field are absent.

**Comment by justinsb** [comment]
Thanks @acpana 

/approve
/lgtm

**Pull request merged**